### PR TITLE
Added error handling for case where EGM port for a given mech unit group is missing

### PR DESCRIPTION
**Pull request by stephanie-eng**
Add a helpful hint when trying to run the driver without specifying the EGM port for a mech unit group.
In the example below, we try to launch without specifying the hardware parameter "extaxegm_port"
image

### CODE REVIEW

File: abb_hardware_interface/src/abb_system_position_only.cpp [hunk 1]
```diff
@@ -82,12 +82,18 @@ CallbackReturn ABBSystemPositionOnlyHardware::on_init(const hardware_interface::
...
-                                << group.name() << " on port " << egm_port);
+    try{
+      const auto egm_port = stoi(info_.hardware_parameters[group.name() + "egm_port"]);
+      const auto channel_configuration =
+        abb::robot::EGMManager::ChannelConfiguration{static_cast<uint16_t>(egm_port), group};
+      channel_configurations.emplace_back(channel_configuration);
+      RCLCPP_INFO_STREAM(LOGGER, "Configuring EGM for mechanical unit group "
+                                  << group.name() << " on port " << egm_port);
+    }
+    catch (std::invalid_argument & e){
```
Comment (AndyZe) [inline abb_hardware_interface/src/abb_system_position_only.cpp:93]: It's good practice to minimize what's in the try{}. Here, if I understand right, it should only be one line.

```suggestion
    const auto egm_port = stoi(info_.hardware_parameters[group.name() + "egm_port"]);
    try{
      const auto channel_configuration =
        abb::robot::EGMManager::ChannelConfiguration{static_cast<uint16_t>(egm_port), group};
    }
    catch (std::invalid_argument & e){
      ...
    }
    channel_configurations.emplace_back(channel_configuration);
    RCLCPP_INFO_STREAM(LOGGER, "Configuring EGM for mechanical unit group "
                                  << group.name() << " on port " << egm_port);
```

Comment (stephanie-eng) [reply abb_hardware_interface/src/abb_system_position_only.cpp:93]: That was what I initially tried, but it didn't work: the try block has a scope, so we can't access `egm_port` and `channel_configuration` outside of the try block.

Comment (AndyZe) [reply abb_hardware_interface/src/abb_system_position_only.cpp:93]: Well, it looks like you can at least move one more line out of the the `try{}`:

```
    const auto egm_port = stoi(info_.hardware_parameters[group.name() + "egm_port"]);
    try{
      const auto channel_configuration =
        abb::robot::EGMManager::ChannelConfiguration{static_cast<uint16_t>(egm_port), group};
      channel_configurations.emplace_back(channel_configuration);
      RCLCPP_INFO_STREAM(LOGGER, "Configuring EGM for mechanical unit group "
                                  << group.name() << " on port " << egm_port);
    }
    catch (std::invalid_argument & e){
      RCLCPP_FATAL_STREAM(LOGGER, "EGM port for mechanical unit group \"" << group.name() << "\" not specified in hardware parameters");
      return CallbackReturn::ERROR;
    }
```

Comment (stephanie-eng) [reply abb_hardware_interface/src/abb_system_position_only.cpp:93]: Unfortunately, that first line is the one that will error, so it has to be in the `try{}`

Comment (AndyZe) [reply abb_hardware_interface/src/abb_system_position_only.cpp:93]: Hmm, I built that code before posting it. You mean it errors at runtime?

Comment (AndyZe) [reply abb_hardware_interface/src/abb_system_position_only.cpp:93]: Oh, I understand. OK, no problem

**Review by AndyZe — APPROVED** [review]

**Pull request merged**